duracloud/duracloud
# Dependency updates for Spring 5 support

**JIRA Ticket**: ?

# What does this Pull Request do?

## Dependency Updates

* Spring -> 5.3.27
* Spring Security -> 5.8.3
* Tiles -> 3.0.1

+ other minor bumps

## Config Changes

* Removed version from bean schema location where possible
* Moved non-duracloud beans to java config
  * This was to help with updating packages, default parameters, etc, as I wasn't getting hints in the xml. 

## Spring Security

* Use SecurityContextHolderStrategy when retrieving SecurityContext
* Removed setting of SecurityContextHolder strategy
  * This is something which broke method security in the management-console, so I replicated the changes here as well though I'm not as certain if they're necessary. In addition, the SecurityContextHolderStrategy is retrieved first so this should be ok.
* Configure an additional `NoOpPasswordEncoder` for durastore
  * @dbernstein this is what I had mentioned on our call the other week. This allows the http calls being made from duradmin to complete successfully but is something we'll want to take another look at. If you want I can get some more info, I forget exactly what class I was stepping through but I know it was hitting `/stores`.
* Minor updates to move to MessageDigestPasswordEncoder

## Tiles

* API changes for updates to Tiles 3
* Moved views.xml config to be in default location for the ViewResolver
* Removed old AbstractSpringPreparerFactory
  * @dbernstein this is the other thing I had mentioned last week. As far as I can tell it wasn't being used in any way, and the application still deploys and is able to be used from my brief testing.

## Other

* Removed logging from BitIntegrityReportRest constructor
  * Since the object being logged is a request scoped proxy, this was throwing an exception in the Tomcat logs on startup. In favor of reducing noise I opted to remove this but could always add it back without the proxy object.
* Various minor API updates, e.g. `PageRequest.of` 

# How should this be tested?

* Verify the project still builds
* Deploy into Tomcat and startup (with no errors)
* Run integration tests please and thank you

# Additional Notes:

I haven't run the integration tests yet, I've been meaning to try but need to verify my environment is setup correctly first.

# Interested parties
@dbernstein
